soc/xtensa/intel_adsp/tools/cavstool_client.py

```diff
@@ -41,27 +43,57 @@ def send_cmd(self, cmd):
 
             if ack == CMD_LOG_START:
                 self.monitor_log()
-            elif ack == CMD_LOG_STOP:
-                log.info(f"Stop output.")
             elif ack == CMD_DOWNLOAD:
                 self.run()
             else:
                 log.error(f"Receive incorrect msg:{ack} expect:{cmd}")
 
-    def download(self, filename):
+    def uploading(self, filename):
         # Send the FW to server
+        fname = os.path.basename(filename)
+        fsize = os.path.getsize(filename)
+        flen = len(fname)
+
+        md5_tx = hashlib.md5(open(filename,'rb').read()).hexdigest()
+
+        # Pack the header and the expecting packed size is 80 bytes.
+        # The header by convention includes:
+        # size(4), filename(42), MD5(32) and pad(2)
+        values = (fsize, fname.encode('utf-8'), md5_tx.encode('utf-8'), flen)
+        log.info(f'filename:{fname}, size:{fsize}, md5:{md5_tx}, len {flen}')
+
+        s = struct.Struct('I 42s 32s H')
```

Comment:
""convention" sounds like it's defined somewhere else but `'I 42s 32s H'` is defined only in this repo (which is great, it means it can be changed any time without breaking backward compatibility).

Please name this "magic value" / protocol header a name at the top of the file:

```
PACKET_HEADER_FORMAT_SOMETHING = 'I 42s 32s H'
```

Reply:
Good suggestion got it fixed. Thanks!